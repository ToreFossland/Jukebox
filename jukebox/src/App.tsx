import React, {} from 'react';
import './resources/styling/layout.css';
import './App.css';
import NavBar from "./components/navbar";
import Home from "./components/home";
import Info from "./components/info";
import Footer from "./components/footer";
import Player from "./components/player";
import Dancers from "./components/dancers";
import {Provider} from './context'

interface myState {
  trackID: number,
  btheme: string,
  ctheme: string,
  bodyTheme: string,
  dancerIndex: number
}

interface myProps {
//fill in props here
}

class App extends React.Component<myProps, myState> {
  constructor(props:any){
    super(props);
    this.state = {
        trackID: 0,
        btheme: "#fff",
        bodyTheme: "#eee",
        ctheme: "#333",
        dancerIndex: 0
    }
    this.changeTheme = this.changeTheme.bind(this);
    this.changeDancer = this.changeDancer.bind(this);
}

  componentDidMount() {
    let btheme = localStorage.getItem('btheme');
    let ctheme = localStorage.getItem('ctheme');
    let bodyTheme = localStorage.getItem('bodyTheme');
    if(btheme != null) {
      this.setState({btheme: btheme});
    }

    if(ctheme != null) {
      this.setState({ctheme: ctheme});
    }

    if(bodyTheme != null) {
      this.setState({bodyTheme: bodyTheme});
    }

    let dancerIndex = sessionStorage.getItem('dancerIndex');

    if(dancerIndex != null) {
      this.setState({dancerIndex: parseInt(dancerIndex)});
    }

  }
  changeTheme() {
    const newbTheme = this.state.btheme == "#fff" ? "#555" : "#fff";
    const newcTheme = this.state.ctheme == "#333" ? "#eee" : "#333";
    const newBodyTheme = this.state.bodyTheme == "#111" ? "#eee" : "#111";
    this.setState({ctheme: newcTheme});
    this.setState({bodyTheme: newBodyTheme});
    this.setState({btheme: newbTheme});
    localStorage.setItem('btheme', newbTheme);
    localStorage.setItem('ctheme', newcTheme);
    localStorage.setItem('bodyTheme', newBodyTheme);
  }

  changeDancer(index: number) {
    this.setState({dancerIndex: index})
    sessionStorage.setItem('dancerIndex', JSON.stringify(index));
  }


  render(){
    return(
      <Provider>
        <div className="App" style={{background: this.state.bodyTheme}}>
          <NavBar/>
          <div id="main">
            <Home dancerIndex={this.state.dancerIndex} bordertheme={this.state.bodyTheme}  btheme={this.state.btheme} ctheme={this.state.ctheme}/>
            <Dancers btheme={this.state.btheme} bordertheme={this.state.bodyTheme} ctheme={this.state.ctheme} onChange={this.changeDancer} />
            <Info btheme={this.state.btheme} ctheme={this.state.ctheme}/>
          </div>
          <Footer onChange={this.changeTheme}/>
          <Player/>
        </div>
      </Provider>
      

    );
  }
}


export default App;
