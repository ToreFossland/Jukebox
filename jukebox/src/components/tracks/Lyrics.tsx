import React, {Component} from "react";
import axios from 'axios';
import Track from "./Track";
import Spinner from "../../resources/media/Spinner";


class Lyrics extends Component{

    state = {
        track: {track: undefined},
        lyrics: {}
    }


    componentDidMount() {
        axios.get(`https://cors-anywhere.herokuapp.com/https://api.musixmatch.com/ws/1.1/track.lyrics.get?track_id=190589562&apikey=${process.env.REACT_APP_MM_KEY}`)
            .then(res => {
                console.log(res.data);
                this.setState({ lyrics: res.data.message.body.lyrics_body});

                return axios.get(`https://cors-anywhere.herokuapp.com/https://api.musixmatch.com/ws/1.1/track.get?commontrack_id=5920049&apikey=${process.env.REACT_APP_MM_KEY}`)
            })
            .then(res => {
                this.setState({track: res.data.message.body.track})
            })
            .catch(err => console.log(err))
    }

    render() {
        const {track, lyrics} = this.state;
        if( track === undefined ||
            lyrics === undefined ||
            Object.keys(track).length === 0 ||
            Object.keys(lyrics).length === 0){
            return <Spinner />
        }else{
            return <h1>data returned</h1>
        }
    }
}

export default Lyrics;