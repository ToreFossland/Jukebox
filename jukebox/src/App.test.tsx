import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer'
import App from './App';
import Songs from './components/songs';
import Home from './components/home';
import Info from './components/info';
import Dancers from './components/dancers';
import NavBar from './components/navbar';
import Footer from './components/footer';
import Player from './components/player';
import Lyrics from './components/lyrics';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

it('renders correctly', () => {
    const tree = renderer
        .create(<App />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders correctly', () => {
    const tree = renderer
        .create(<Songs />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
        .create(<Home />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
        .create(<Info />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
        .create(<Dancers />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
        .create(<NavBar />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
        .create(<Footer />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
        .create(<Player />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
        .create(<Lyrics />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});