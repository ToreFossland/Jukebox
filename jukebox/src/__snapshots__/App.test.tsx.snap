// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
  style={
    Object {
      "background": "#eee;",
    }
  }
>
  <div
    id="navbar"
  >
    <div
      className="navelements"
    >
      <a
        href="#main"
        id="jukeboxbutton"
      >
        <img
          alt="Img"
          src="jukebox_logo_bigger.png"
        />
        <p>
          JukeBox
        </p>
      </a>
      <a
        className="navbuttons"
        href="#songmain"
      >
        <p>
          Songs
        </p>
      </a>
      <a
        className="navbuttons"
        href="#dancermain"
      >
        <p>
          Dancers
        </p>
      </a>
      <a
        className="navbuttons"
        href="#infomain"
      >
        <p>
          Info
        </p>
      </a>
    </div>
  </div>
  <div
    id="main"
  >
    <div
      id="container3"
      style={
        Object {
          "background": "#fff",
          "color": "#333",
        }
      }
    >
      <svg
        id="svgBackground"
      >
        <defs>
          <linearGradient
            id="gradient-0"
            x1="0%"
            x2="0%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stopColor="#DB3A32"
              stopOpacity="1"
            />
            <stop
              offset="50%"
              stopColor="#FAD500"
              stopOpacity="1"
            />
            <stop
              offset="100%"
              stopColor="#FAD500"
              stopOpacity="0"
            />
          </linearGradient>
        </defs>
        <rect
          fill="url(#gradient-0)"
          height="100%"
          width="100%"
        />
      </svg>
      <div
        id="homemain"
        style={
          Object {
            "borderColor": "#eee;",
          }
        }
      >
        <div
          id="homedefault"
        >
          <h1>
            Welcome to JukeBox
          </h1>
          <p>
             Here we have a wide selection of songs available for karaoke usage. Click "Songs" in the menu or scroll down to begin!
          </p>
        </div>
        <div
          id="songmain"
        >
          <h1>
            Songs
          </h1>
          <p>
            Here is our library of available songs.
          </p>
          <div
            className="songlist"
          />
        </div>
      </div>
    </div>
    <div
      id="dancermain"
      style={
        Object {
          "background": "#fff",
          "borderColor": "#eee;",
          "color": "#333",
        }
      }
    >
      <h1>
        Dancers
      </h1>
      <p>
        Click on the dancer you want to see on top of the screen!
      </p>
      <div
        className="dancerlist"
      >
        <div>
          <img
            onClick={[Function]}
            src="dancer1.png"
          />
          <p>
            Effect: clickable!
          </p>
        </div>
        <div>
          <img
            onClick={[Function]}
            src="dancer2.png"
          />
          <p>
            Effect: clickable!
          </p>
        </div>
        <div>
          <img
            onClick={[Function]}
            src="dancer3.png"
          />
          <p>
            Effect: clickable!
          </p>
        </div>
        <div>
          <img
            onClick={[Function]}
            src="dancer4.png"
          />
          <p>
            Effect: hoverable!
          </p>
        </div>
        <div>
          <img
            onClick={[Function]}
            src="dancer5.png"
          />
          <p>
            Effect: hoverable!
          </p>
        </div>
        <div>
          <img
            onClick={[Function]}
            src="dancer6.png"
          />
          <p>
            Effect: hoverable!
          </p>
        </div>
      </div>
    </div>
    <div
      id="infomain"
      style={
        Object {
          "background": "#fff",
          "color": "#333",
        }
      }
    >
      <h1>
        Info
      </h1>
      <p>
         Press Dua Lipa's head, torso or tonga to change their color!
      </p>
    </div>
  </div>
  <div
    id="footermain"
  >
    <h2>
      Created By:
    </h2>
    <p>
      Haakon, Kaspar og Tore.
    </p>
    <button
      id="pageThemeButton"
      onClick={[Function]}
    >
      Change Theme
    </button>
  </div>
  <div
    id="player"
  >
    <div
      className="playerElements"
    >
      <button
        className="playerBtns"
        onClick={[Function]}
      >
        <img
          id="playbutton"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "play.svg",
            }
          }
        />
      </button>
      <div
        className="playerTime"
      >
        <p>
          0:00
        </p>
      </div>
      <div
        className="progressBar"
      >
        <input
          className="invisibleSlider"
          max="100"
          min="0"
          onClick={[Function]}
          type="range"
        />
        <div
          className="progressDot"
          style={
            Object {
              "marginLeft": "-1%",
            }
          }
        />
        <div
          id="progressLineBackground"
        >
          <div
            id="progressLine"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
      </div>
      <div
        className="playerTime"
      >
        <p>
          0:00
        </p>
      </div>
      <div
        className="playerSongImg"
        style={
          Object {
            "backgroundImage": "url(0.jpg)",
          }
        }
      />
      <div
        className="playerSongData"
      >
        <p
          id="playerTrackName"
        >
          
        </p>
        <p
          id="playerTrackAlbum"
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  id="songmain"
>
  <h1>
    Songs
  </h1>
  <p>
    Here is our library of available songs.
  </p>
  <div
    className="songlist"
  />
</div>
`;

exports[`renders correctly 3`] = `
<div
  id="infomain"
  style={
    Object {
      "background": undefined,
      "color": undefined,
    }
  }
>
  <h1>
    Info
  </h1>
  <p>
     Press Dua Lipa's head, torso or tonga to change their color!
  </p>
</div>
`;

exports[`renders correctly 4`] = `
<div
  id="dancermain"
  style={
    Object {
      "background": undefined,
      "borderColor": undefined,
      "color": undefined,
    }
  }
>
  <h1>
    Dancers
  </h1>
  <p>
    Click on the dancer you want to see on top of the screen!
  </p>
  <div
    className="dancerlist"
  >
    <div>
      <img
        onClick={[Function]}
        src="dancer1.png"
      />
      <p>
        Effect: clickable!
      </p>
    </div>
    <div>
      <img
        onClick={[Function]}
        src="dancer2.png"
      />
      <p>
        Effect: clickable!
      </p>
    </div>
    <div>
      <img
        onClick={[Function]}
        src="dancer3.png"
      />
      <p>
        Effect: clickable!
      </p>
    </div>
    <div>
      <img
        onClick={[Function]}
        src="dancer4.png"
      />
      <p>
        Effect: hoverable!
      </p>
    </div>
    <div>
      <img
        onClick={[Function]}
        src="dancer5.png"
      />
      <p>
        Effect: hoverable!
      </p>
    </div>
    <div>
      <img
        onClick={[Function]}
        src="dancer6.png"
      />
      <p>
        Effect: hoverable!
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly 5`] = `
<div
  id="navbar"
>
  <div
    className="navelements"
  >
    <a
      href="#main"
      id="jukeboxbutton"
    >
      <img
        alt="Img"
        src="jukebox_logo_bigger.png"
      />
      <p>
        JukeBox
      </p>
    </a>
    <a
      className="navbuttons"
      href="#songmain"
    >
      <p>
        Songs
      </p>
    </a>
    <a
      className="navbuttons"
      href="#dancermain"
    >
      <p>
        Dancers
      </p>
    </a>
    <a
      className="navbuttons"
      href="#infomain"
    >
      <p>
        Info
      </p>
    </a>
  </div>
</div>
`;

exports[`renders correctly 6`] = `
<div
  id="footermain"
>
  <h2>
    Created By:
  </h2>
  <p>
    Haakon, Kaspar og Tore.
  </p>
  <button
    id="pageThemeButton"
    onClick={[Function]}
  >
    Change Theme
  </button>
</div>
`;

exports[`renders correctly 7`] = `
<div
  id="lyricsmain"
>
  <h1>
    Lyrics
  </h1>
  <div
    id="lyricsText"
  >
    <p
      id="hore"
    >
      
    </p>
  </div>
</div>
`;
